packages/codegen/src/generate.ts

```diff
@@ -447,11 +457,12 @@ export default class ApiGenerator {
 
     if (!this.refs[$ref]) {
       let schema = this.resolve<SchemaObject>(obj);
-      if (ignoreDiscriminator) {
+      if (typeof schema !== "boolean" && ignoreDiscriminator) {
         schema = _.cloneDeep(schema);
         delete schema.discriminator;
       }
-      const name = schema.title || getRefName($ref);
+      const name =
+        (typeof schema !== "boolean" && schema.title) || getRefName($ref);
```

Comment:
Can we do an early return in case the schema is bool? Would need to check what to return in that case.

Reply:
Not sure how that would look like. The method technically just returns a `ts.TypeNode`, but it also some side effects, mutating `this.refs` and `this.aliases`.

Reply:
Yeah, might be that this is not possible - will have a look

Reply:
Added a commit - could you take a look?

Reply:
I added an (unfinished) test for referenced boolean schemas, but it errors:
```
TypeError: Cannot read properties of undefined (reading 'base')
 ❯ ApiGenerator.getRefAlias codegen/src/generate.ts:462:32
    460| 
    461|       if (typeof schema === "boolean") {
    462|         return this.refs[$ref].base;
       |                                ^
    463|       }
```

Reply:
I pushed a fix (722bbde) but it repeats some code. Do you think we should refactor this a bit more?

Reply:
Nice work! I think we should go back to what you initially had. Compared to the current solution it's way less code and repetition and it also actually works. 
Sorry for the back and forth, I'll add a change that keeps your great tests and moves the implementation back to what you initially proposed.

Reply:
No worries. Didn't spend a lot of time on this at all.

Reply:
Added it in. What do you think?

Reply:
LGTM 👍🏼